### PR TITLE
Documenation updates

### PR DESCRIPTION
**Pull request by mrsabath**

### CODE REVIEW

**Comment by mrsabath** [comment]
Referencing #113 issue. 
Fixing #112 issue.
Referencing #107 
Fixing #105

File: charts/spire/templates/k8s-workload-registrar-configmap.tpl [hunk 1]
```diff
@@ -10,7 +10,7 @@ data:
     trust_domain = "{{ .Values.trustdomain }}"
     # enable when direct socket access to SPIRE Server available:
     # server_socket_path = "/run/spire/sockets/registration.sock"
-    agent_socket_path = "/run/spire/sockets/agent.sock"
+    agent_socket_path = "{{ .Values.agentSocketDir }}/{{ .Values.agentSocketFile }}"
```
Comment (lumjjb) [inline charts/spire/templates/k8s-workload-registrar-configmap.tpl:13]: nit, but would it be a bit cleaner to do nested values instead, may lead to easier to navigate `values.yaml` file as well

`{{ .Values.agent.socketDir }}`

Comment (lumjjb) [reply charts/spire/templates/k8s-workload-registrar-configmap.tpl:13]: ah you do that for oidc, maybe we can follow that pattern

File: charts/tornjak/templates/server-statefulset.tpl [hunk 2]
```diff
@@ -43,56 +42,60 @@ spec:
             - name: certs
               mountPath: /opt/spire/sample-keys
             - name: spire-server-socket
-              mountPath: /run/spire/sockets
+              mountPath: {{ .Values.spireServerSocketDir }}
               readOnly: false
-{{- if .Values.k8s_psat.remoteClusters }}
+            {{- if .Values.k8s_psat.remoteClusters }}
```
Comment (lumjjb) [inline charts/tornjak/templates/server-statefulset.tpl:47]: maybe can make this `attestors.k8s_psat`

File: docs/spire-multi-cluster.md [hunk 3]
```diff
@@ -55,6 +56,23 @@ Procedures for obtaining these values are [here](https://docs.aws.amazon.com/gen
...
+
+Or add the `--set "aws=true"` flag to the helm command:
+```console
+helm install --set "spireServer.address=$SPIRE_SERVER" \
+--set "spireServer.port=$SPIRE_PORT"  --set "namespace=$AGENT_NS" \
+--set "clustername=$CLUSTERNAME" --set "region=us-east" \
+--set "trustdomain=openshift.space-x.com" \
+--set "aws=true" \
+spire charts/spire --debug
+```
```
Comment (lumjjb) [inline docs/spire-multi-cluster.md:74]: probably in the future, we'd need a subconfiguration for custom metadata endpoint, but should be ok for most deployments, and can support it if its asked for

File: charts/spire/values.yaml [hunk 4]
```diff
@@ -1,31 +1,45 @@
...
+# SPIRE Agent related configuration
+spireAgent:
+  img: gcr.io/spiffe-io/spire-agent
+  # SPIRE Agent socket:
+  socketDir: /run/spire/sockets
+  socketFile: agent.sock
+
+# Workload Registrar configuration
+spireRegistrar:
+  img: gcr.io/spiffe-io/k8s-workload-registrar
```
Comment (lumjjb) [inline charts/spire/values.yaml:36]: probably would be nice here to be able to specify the identity template and label configuration here (using our defaults)

File: charts/tornjak/templates/server-statefulset.tpl (hunk 2 above)
Comment (mrsabath) [reply charts/tornjak/templates/server-statefulset.tpl]: done

File: charts/spire/values.yaml (hunk 4 above)
Comment (mrsabath) [reply charts/spire/values.yaml:36]: This is a nice suggestion, but it is a bit challenging if we want to include `context` values that are dynamic and depend on the other helm variables. This should be investigated further, so I created a new issue #115

File: docs/spire-multi-cluster.md (hunk 3 above)
Comment (mrsabath) [reply docs/spire-multi-cluster.md:74]: Captured in issue #116

**Review by lumjjb — APPROVED** [review]
lgtm

**Pull request merged**